### PR TITLE
[NFC] Remove ModuleDecl::isClangModule in favor of isNonSwiftModule.

### PR DESCRIPTION
**Pull request by mateioprea**
Fixes SR-13237. (https://bugs.swift.org/browse/SR-13237)

### CODE REVIEW

**Comment by mateioprea** [comment]
@varungandhi-apple can you help with reviewing this? Thanks

**Comment by CodaFi** [comment]
@swift-ci smoke test

File: cmake/modules/DarwinSDKs.cmake
Comment (varungandhi-apple) [inline cmake/modules/DarwinSDKs.cmake]: This shouldn't need to be committed, it is a local workaround. Hopefully there is a new CMake release soon with the fix: https://gitlab.kitware.com/cmake/cmake/-/merge_requests/5023

File: lib/AST/Module.cpp
Comment (varungandhi-apple) [inline lib/AST/Module.cpp:498]: The corresponding method declaration in `include/swift/AST/Module.h` should also be deleted.

**Comment by mateioprea** [comment]
That means that I should remake the PR, right?

**Comment by varungandhi-apple** [comment]
You could do an interactive rebase (drop the second commit, modify the first commit to update the header) and then force-push.

**Comment by CodaFi** [comment]
We can also squash-and-merge this for you.

**Comment by mateioprea** [comment]
@CodaFi @varungandhi-apple done 👍

**Review by varungandhi-apple — APPROVED** [review]
Thanks for the cleanup!

**Comment by varungandhi-apple** [comment]
@swift-ci please smoke test and merge

**Comment by varungandhi-apple** [comment]
@swift-ci please smoke test and merge

**Comment by CodaFi** [comment]
@swift-ci smoke test macOS

**Comment by CodaFi** [comment]
@swift-ci smoke test macOS

**Comment by mateioprea** [comment]
"master" branch was also failing yesterday. Should I rebase and run the smoke tests again?

**Comment by varungandhi-apple** [comment]
master seems fine. The Java exceptions are related to some Jenkins + GitHub voodoo, where it doesn't fetch things properly from GitHub or it fetches out-of-date stuff.

@swift-ci please smoke test macOS platform

**Comment by mateioprea** [comment]
Ok. Do we need to do something here then? Or we should trigger the tests until they pass?

**Comment by varungandhi-apple** [comment]
That usually doesn't happen so often. I'm seeing another test fail, let me ask around. If it keeps failing, it needs to be x-failed, doesn't make sense to keep triggering the tests repeatedly.

**Comment by varungandhi-apple** [comment]
This PR should fix the issue we hit here: https://github.com/apple/llvm-project/pull/1576

Let's wait to re-run tests until that gets merged.

**Comment by varungandhi-apple** [comment]
The LLVM PR apple/llvm-project#1576 is going to take some time to get fixed because there is a more serious issue than just cherry-picking. We probably shouldn't stay blocked on that.

@swift-ci please smoke test macOS

**Comment by mateioprea** [comment]
I see the OS X tests are failing. Maybe we should wait for https://github.com/apple/llvm-project/pull/1576 ?

**Comment by varungandhi-apple** [comment]
@swift-ci please smoke test macOS

**Pull request merged**
